kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -104,6 +111,14 @@ public Transaction build(Engine engine) {
         boolean isNewTable = snapshot.getVersion(engine) < 0;
         validate(engine, snapshot, isNewTable);
 
+        Metadata  metadata = snapshot.getMetadata().clone();
+        if (tableProperties.isPresent()) {
+            TableConfig.validateProperties(tableProperties.get());
+            Map<String, String> newConfiguration = new HashMap<>(metadata.getConfiguration());
+            newConfiguration.putAll(tableProperties.get());
```

Comment:
Shouldn't we checking if the property already exists and can it be replaced? What does the Delta-Spark do?

Reply:
I think[ Delta-Spark](https://github.com/delta-io/delta/blob/fa7d6c0eb5de0e05b0efac6b11cf96c58740a828/spark/src/main/scala/org/apache/spark/sql/delta/commands/alterDeltaTableCommands.scala#L157) replaces it for most of the cases.